libsimulator/src/SocialForceModel.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "CollisionFreeSpeedModelData.hpp"
+#include "CollisionGeometry.hpp"
+#include "NeighborhoodSearch.hpp"
+#include "OperationalModel.hpp"
+#include "UniqueID.hpp"
+
+struct GenericAgent;
+
+class SocialForceModel : public OperationalModel
+{
+public:
+    using NeighborhoodSearchType = NeighborhoodSearch<GenericAgent>;
+
+private:
+    double _cutOffRadius{2.5};
+    double bodyForce;
+    double friction;
+
+public:
+    SocialForceModel(double bodyForce_, double friction_);
+    ~SocialForceModel() override = default;
+    OperationalModelType Type() const override;
+    OperationalModelUpdate ComputeNewPosition(
+        double dT,
+        const GenericAgent& ped,
+        const CollisionGeometry& geometry,
+        const NeighborhoodSearchType& neighborhoodSearch) const override;
+    void ApplyUpdate(const OperationalModelUpdate& update, GenericAgent& agent) const override;
+    void CheckModelConstraint(
+        const GenericAgent& agent,
+        const NeighborhoodSearchType& neighborhoodSearch,
+        const CollisionGeometry& geometry) const override;
+    std::unique_ptr<OperationalModel> Clone() const override;
+
+    Point drivingForce(const GenericAgent& agent) const;
+    Point AgentForce(const GenericAgent& ped1, const GenericAgent& ped2) const;
+    Point ObstacleForce(const GenericAgent& agent, const LineSegment& segment) const;
+    double pushing_Force(double A, double B, double r, double distance) const;
```

Comment:
DrivingForce and PushingForceLength(old name: pushing_Force) are now static
AgentForce and ObstacleForce currently remain const as they use the model parameters which are part the class.
It would however also be possible to make them static anyway.